requirements.txt

```diff
@@ -2,12 +2,13 @@ altair>=4.0.0,<5
 Click>=7.1.2
 colorama>=0.4.3
 cryptography>=3.2,<4.0.0
+dataclasses
 importlib-metadata>=1.7.0 # (included in Python 3.8 by default.)
-ipywidgets>=7.5.1
+Ipython>=7.16.3
```

Comment:
Are the additions of dataclasses and Ipython intended? Could you provide a bit more context?

Reply:
Absolutely; my bad! They were both undeclared dependencies being brought in elsewhere.

ipython is used in for a `core/util.py` method  that `ExpectationValidationResult` uses;
dataclasses is part of the core library from python 3.7 on, but must be explicitly installed for 3.6 🙃